src/LibraryViewExtensionMSWebBrowser/LibraryViewController.cs

```diff
@@ -0,0 +1,583 @@
+            interop = new scriptingObject(this);
+        }
+
+        //if the window is resized toggle visibility of browser to force redraw
+        private void DynamoWindow_SizeChanged(object sender, SizeChangedEventArgs e)
+        {
+            if (browser != null)
+            {
+                browser.InvalidateVisual();
+            }
+        }
+
+        //if the dynamo window is minimized and then restored, force a layout update.
+        private void DynamoWindowStateChanged(object sender, EventArgs e)
+        {
+            if (browser != null)
+            {
+                browser.InvalidateVisual();
+            }
+        }
+
+        #region methods exposed to JS
+        /// <summary>
+        /// Call this method to create a new node in Dynamo canvas.
+        /// </summary>
+        /// <param name="nodeName">Node creation name</param>
+        public void CreateNode(string nodeName)
+        {
+            dynamoWindow.Dispatcher.BeginInvoke(new Action(() =>
+            {
+                //if the node we're trying to create is a customNode, lets disable the eventObserver.
+                // this will stop the libraryController from refreshing the libraryView on custom node creation.
+                var resultGuid = Guid.Empty;
+                if (Guid.TryParse(nodeName, out resultGuid))
+                {
+                    this.disableObserver = true;
+                }
+                //Create the node of given item name
+                var cmd = new DynamoModel.CreateNodeCommand(Guid.NewGuid().ToString(), nodeName, -1, -1, true, false);
+                commandExecutive.ExecuteCommand(cmd, Guid.NewGuid().ToString(), LibraryViewExtensionMSWebBrowser.ExtensionName);
+                LogEventsToInstrumentation(CreateNodeInstrumentationString, nodeName);
+
+                this.disableObserver = false;
+            }));
+        }
+
+        /// Call this method to import a zero touch library. It will prompt
+        /// user to select the zero touch dll.
+        /// </summary>
+        public void ImportLibrary()
+        {
+            dynamoWindow.Dispatcher.BeginInvoke(new Action(() =>
+                dynamoViewModel.ImportLibraryCommand.Execute(null)
+            ));
+        }
+        /// <summary>
+        /// This function logs events to instrumentation if it matches a set of known events
+        /// </summary>
+        /// <param name="eventName">Event Name that gets logged to instrumentation</param>
+        /// <param name="data"> Data that gets logged to instrumentation </param>
+        public void LogEventsToInstrumentation(string eventName, string data)
+        {
+            if (eventName == "Search" || eventName == "Filter-Categories" || eventName == "Search-NodeAdded")
+            {
+                Analytics.LogPiiInfo(eventName, data);
+            }
+        }
+        internal void refreshLibraryView(WebBrowser browser)
```

Comment:
called by javascript side, does a full refresh of the loadedTypes and LayoutSpec - this actually quite fast and is not called frequently. ( < 50ms in my tests with several large packages installed)